src/context/TransactionsProvider/TransactionsProvider.tsx

```diff
@@ -55,6 +58,31 @@ export const TransactionsProvider: React.FC<TransactionsProviderProps> = ({ chil
 
       const { getOpportunitiesUserData } = opportunitiesApi.endpoints
 
+      const idleContractAddresses = [
+        ...Object.values(stakingOpportunitiesById)
```

Comment:
This could be a selector, but feels weird to shoehorn Idle-specific stuff into generalized staking selectors. This is O(n) where n is not that many opportunities, and the only thing we're doing inside the iteration is accessing a property. All of that only done when a Tx is confirmed, so this is probably a good home?

Reply:
I'm all for keeping specific things like this out of selectors 👌